protocol/avdl/chat_local.avdl

```diff
@@ -56,17 +57,28 @@ protocol chatLocal {
 
     union { null, Time } After;
     union { null, Time } Before;
-    bool onlyNew;
-    int limitNumber;
+    boolean onlyNew;
+    
+    int limitPerConversation;
+    int limitOfConversations;
+
+    array<chat1.ConversationID> conversations;
+
+    boolean markAsRead;
+  }
+
+  record ConversationMessagesLocal {
+    chat1.ConversationID id;
+    array<Message> messages;
   }
 
   chat1.InboxView getInboxLocal(union { null, chat1.Pagination } pagination);
-  ThreadView getThreadLocal(chat1.ConversationID conversationID, union { null, chat1.Pagination } pagination);
+  ThreadView getThreadLocal(chat1.ConversationID conversationID, boolean markAsRead, union { null, chat1.Pagination } pagination);
   void postLocal(chat1.ConversationID conversationID, MessagePlaintext messagePlaintext);
   chat1.ConversationID newConversationLocal(chat1.ConversationIDTriple conversationTriple);
 
   chat1.ConversationID getOrCreateTextConversationLocal(string tlfName, string topicName, chat1.TopicType topicType);
-  array<Message> getMessagesLocal(MessageSelector selector);
+  array<ConversationMessagesLocal> getMessagesLocal(MessageSelector selector);
```

Comment:
I've been experimenting myself with different versions of the interface between CLI and the service. I think the current `getMessagesLocal` should cover most message fetching logics from CLI. Let me know if this looks reasonable!